Detectors/FIT/FT0/workflow/include/FT0Workflow/FT0DataReaderDPLSpec.h

```diff
@@ -0,0 +1,63 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   FT0DataReaderDPLSpec.h
+
+#ifndef O2_FT0DATAREADERDPLSPEC_H
+#define O2_FT0DATAREADERDPLSPEC_H
+
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Task.h"
+#include "Framework/CallbackService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Output.h"
+#include "Framework/WorkflowSpec.h"
+#include "Framework/SerializationMethods.h"
+#include "DPLUtils/DPLRawParser.h"
+#include "FT0Raw/DataBlockReader.h"
+#include "FT0Raw/DigitBlockFT0.h"
+
+#include "DataFormatsFT0/Digit.h"
+#include "DataFormatsFT0/ChannelData.h"
+#include <iostream>
+#include <vector>
+#include <gsl/span>
+using namespace o2::framework;
+
+namespace o2
+{
+namespace ft0
+{
+template <bool IsExtendedMode>
```

Comment:
What is the reason to make this template parameter? As far as I can see, the isExtendedMode is used as a usual data member.

Reply:
This parameter is used in the std::conditional<IsExtendedMode, DataBlockTCMext, DataBlockTCM>::type which defines type of TCM data(too modes: normal - 1 TCMdata, extended - 1 TCMdata + 8 TCMextended).

Reply:
ok, and what decides which mode should be used in data procession?

Reply:
Actually this extended mode is about FEE(TCM) settings, by default it will be "normal"(default value is false in the template header). So, in all tests(production performance review) there should be only normal mode(IsExtended = false, default value), extended mode is a technical feature.